python/foxglove-sdk/src/websocket.rs

```diff
@@ -77,22 +94,52 @@ impl ServerListener for PyServerListener {
 
     /// Callback invoked when a client advertises a client channel.
     fn on_client_advertise(&self, client: Client, channel: &ClientChannel) {
-        self.call_client_channel_method(
-            "on_client_advertise",
-            client,
-            channel.id.into(),
-            channel.topic.as_str(),
-        );
+        let client_info = PyClient {
+            id: client.id().into(),
+        };
+
+        let py_channel = PyClientChannel {
+            id: channel.id.into(),
+            topic: channel.topic.clone(),
```

Comment:
I think this does an extra copy before going to a python string, which you could avoid with the PyString type (see `PyChannelView`), but maybe this was intentional since it's infrequently called.

Reply:
Definitely was not intentional since I have no idea what I'm doing when it comes to pyo3. I guess this is enough motivation to change to PyString!